src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -852,6 +948,18 @@ static int client_send_discover(sd_dhcp_client *client) {
                         return r;
         }
 
+        if (client->options && set_size(client->options) > 0) {
```

Comment:
set_size() already returns 0 if the argument is null, so the first part of this conditional is not needed.